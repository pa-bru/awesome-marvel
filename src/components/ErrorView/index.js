import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import Typography from '@material-ui/core/Typography'

import Center from 'components/Center'
import Head from 'components/Head'

const ErrorView = ({ children }) => (
  <Fragment>
    <Head title="Oups..." />
    <Center>
      <Typography variant="body1" color="secondary">
        {children}
      </Typography>
    </Center>
  </Fragment>
)

ErrorView.propTypes = {
  children: PropTypes.node.isRequired,
}

ErrorView.defaultProps = {
  children: 'Something went wrong.',
}

export default ErrorView
