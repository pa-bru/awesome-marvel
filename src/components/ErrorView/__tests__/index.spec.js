import React from 'react'
import { shallow } from 'enzyme'
import ErrorView from '../'

describe('ErrorView', () => {
  it('should render correctly with a default message', () => {
    // When
    const tree = shallow(<ErrorView />)

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('should render with a child', () => {
    // When
    const tree = shallow(<ErrorView>Coucou</ErrorView>)

    // Then
    expect(tree).toMatchSnapshot()
  })
})
