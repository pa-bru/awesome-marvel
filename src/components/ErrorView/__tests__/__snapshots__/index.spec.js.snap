// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView should render correctly with a default message 1`] = `
<React.Fragment>
  <Head
    title="Oups..."
  />
  <Center>
    <WithStyles(Typography)
      color="secondary"
      variant="body1"
    >
      Something went wrong.
    </WithStyles(Typography)>
  </Center>
</React.Fragment>
`;

exports[`ErrorView should render with a child 1`] = `
<React.Fragment>
  <Head
    title="Oups..."
  />
  <Center>
    <WithStyles(Typography)
      color="secondary"
      variant="body1"
    >
      Coucou
    </WithStyles(Typography)>
  </Center>
</React.Fragment>
`;
